controller/controller.go

```diff
@@ -260,6 +300,68 @@ func (c controller) GetStatus(req Request) ([]fleet.UnitStatus, error) {
 	return status, maskAny(err)
 }
 
+func (c controller) WaitForStatus(req Request, desired Status, closer <-chan struct{}) error {
+	fail := make(chan error)
+	done := make(chan struct{})
+
+	go func() {
+		// c describes the count of how often the desired aggregated status was
+		// seen.
+		count := 0
+
+	L1:
+		for {
+			unitStatusList, err := c.groupStatus(req)
+			if IsUnitNotFound(err) && desired == StatusNotFound {
```

Comment:
I think this is a nice trick. It is a trick though. With this we don't need extra methods to handle the destroy cases. Note that there is no status aggregation needed for `StatusNotFound` because there is no state in fleet on destroyed units. Thus we simply expect the not found errors when we want the status to be "not found". 

Another thing. What about naming this `StatusDestroyed`? Not sure if `StatusNotFound` is sufficient.